### ISSUE TITLE
[FEAT] Organise by Stream

### ISSUE THREAD

**Issue by davidwales** [q-masters/vsqlik#281]
**Is your feature request related to a problem? Please describe.**
Currently, all the apps  are placed into a single 'folder' in the VS Code workspace.
Navigation would be improved by dividing this list into folders which reflect the streams in Qlik Sense.

**Describe the solution you'd like**
The structure of the streams in Qlik Sense should be reflected in the folder structure in the VS Code workspace.

**Comment by jheasley322** [comment]
agreed! non-published apps could be in folders by owner - something like "NotPublished.ownername"

**Comment by konne** [comment]
@jheasley322 @davidwales we have discusses this and have decided to go now this approach:

- mywork
    - USERNAME1
      - app FOO
      - app BAR
    - USERNAME2 (if you see apps from multiple users that are not published)
- streams
    - everyone
    - monitoring app
      - app license monitor
      - app site monitor
    - sales

**Comment by r-hannuschka** [comment]
Currently in testing, but i cant make it work like this one:

- mywork
  - USERNAME1
    - app

Since we dont know the name of the user or get no information about who has created the app (own it), the only thing we can made is an QRS Request but the rights of the QMC and Hub are not the same. That means even if i'm admin in the QMC i dont see the apps in hub. But i think this should not be a show stopper for that. 

Currently it looks like that now:
 
![image](https://user-images.githubusercontent.com/25415574/86520761-51f2a280-be48-11ea-9763-dea6fff74ef1.png)

**Comment by konne** [comment]
@davidwales @jheasley322 please test the new release

**Comment by davidwales** [comment]
It works!

**Comment by konne** [comment]
@davidwales please test the new version

**Comment by davidwales** [comment]
@konne It's sort of working.

I can see all the apps in 'my work'.
But I can't see any of the apps under 'streams'.

I wonder if it's because I don't actually have permission on our Qlik server to edit apps in streams?

![image](https://user-images.githubusercontent.com/55477181/88257726-bc895680-cd01-11ea-98d1-8d82831ebb6b.png)

**Comment by konne** [comment]
@r-hannuschka any ideas?

**Comment by r-hannuschka** [comment]
Yeah i check this, but by default i grab all streams via a default operation

```ts
const streams = await global?.getStreamList()

...
const streams = await global?.getStreamList();

if (!streams) {
    return [];
}
...
```

so by default it should open a connection to your qlik sever and we get a stream list (if you have the rights to see them) and if nothing exists it will just return an empty array, which results in a empty directory.

Otherwise we dont have an active connection or no workspace folder but both seems to exists, if not it will also return an empty directory like above.

**Comment by r-hannuschka** [comment]
@davidwales 

I tested now multiple Servers (3 of them) and i see for every one streams, can you please check that you streams on **your_server/hub** and if not then it should be a security problem.

**Comment by davidwales** [comment]
I can see streams and apps in my hub:
![image](https://user-images.githubusercontent.com/55477181/89472141-15a5bf80-d7c3-11ea-8dc9-bed54b561ad7.png)

However, I don't think I have direct edit permissions for these streams.
Perhaps that is the problem?

**Comment by r-hannuschka** [comment]
Mhm not sure about that, by default i simply get all streams which exists maybe i missing something and streams could not be edited on the vscode extensions i disabled this one. So if you see streams on your server i should show them too.

So missing permissions in edit should not be the problem in my optionion? @konne what you say ?

**Comment by davidwales** [comment]
@r-hannuschka From memory, I think I could see all the apps in streams in an earlier version of the extension, so you are probably correct about this.

**Comment by konne** [comment]
@davidwales can we make a skype / teams session together to see the issue?

**Comment by r-hannuschka** [comment]
If i'm not completly wrong, i think qmc get the streams via QRS request and we get them through the websocket which is different. So it is possible the access rights are missing in this case it should be okay not to show them. But on the other side since streams / published apps could not be edited or deleted it could be good to get the informations from them also we can copy out scripts, variables etc from a published app to a non published app.

**Comment by davidwales** [comment]
@konne Sure!

**Comment by konne** [comment]
@davidwales please can you check the new Release 0.9.0

**Comment by davidwales** [comment]
@konne I still can't see any streams in 0.9.0.

**Comment by konne** [comment]
@davidwales what is a good time for you that we have together a look on your system? I can only confirm Ralf, that we use it on different systems without any issue, so I like to understand what makes your system special.
Do you have a chance to invite an Qlik Admin so that we have a chance to check the QMC / Security Rules?

**Comment by davidwales** [comment]
@konne, I've sent a message to our admin. I'll keep you posted.

**Comment by davidwales** [comment]
@konne, Are you free on 7th September at [8 AM Berlin time](https://www.timeanddate.com/worldclock/meetingtime.html?day=7&month=9&year=2020&p1=240&p2=37&iv=0)? (4 PM Sydney time for us).

What email address should we invite to a Teams meeting?

**Comment by konne** [comment]
@davidwales yes would fit. Please use konrad@mattheis.name

**Comment by davidwales** [comment]
@konne, I've sent a Teams invite.

**Comment by davidwales** [comment]
I just tested this on my Mac, and I can see all the streams.
Perhaps this is a Windows only issue?

**Comment by konne** [comment]
@davidwales I think not a direct windows issue. But your finding gives a new idea. Maybe you have an AppName / Streamname that generates on Windows a not allowed Name/Character exception and on Mac not. This could explain the current behavior.

**Comment by r-hannuschka** [comment]
This Information Windows only was a good hint, i have the same error too now. I dont often test in windows because this is only a virtual machine. Now i can start debugging in a windows vm *cheers*.

**Comment by r-hannuschka** [comment]
@konne @davidwales i close this one since this is allready released and open a new issue #502 for that one.
